sycl/source/device_selector.cpp

```diff
@@ -54,18 +55,22 @@ int default_selector::operator()(const device &dev) const {
     }
   }
 
+  // override always wins
+  if (dev.get_info<info::device::device_type>() == detail::get_forced_type())
```

Comment:
I think when the device type is forced we only return devices of the forced type.  Thus, this is redundant.  Can someone who knows the device internals better confirm that?

Reply:
We don't actually do that apparently.  We will limit the devices enumerated to that type _plus the host device_ since the host device must always be available.  In that case, the fact that the host device is available with higher priority means it gets selected.